test/e2e/prometheus_test.go

```diff
@@ -2284,6 +2284,60 @@ func testPromTLSConfigViaSecret(t *testing.T) {
 	}
 }
 
+func testPromStaticProbe(t *testing.T) {
+	t.Parallel()
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+	ctx.SetupPrometheusRBAC(t, ns, framework.KubeClient)
+
+	blackboxExporterName := "blackbox-exporter"
+	if err := framework.CreateBlackBoxExporterAndWaitUntilReady(ns, blackboxExporterName); err != nil {
+		t.Fatal(errors.Wrap(err, "Creating blackbox exporter failed"))
+	}
+
+	blackboxSvc := framework.MakeBlackBoxExporterService(ns, blackboxExporterName)
+	if finalizerFn, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, blackboxSvc); err != nil {
+		t.Fatal(errors.Wrap(err, "creating blackbox exporter service failed"))
+	} else {
+		ctx.AddFinalizerFn(finalizerFn)
+	}
+
+	prometheusName := "test"
+	group := "probe-test"
+	svc := framework.MakePrometheusService(prometheusName, group, v1.ServiceTypeClusterIP)
+
+	proberUrl := blackboxExporterName + ":9115"
+	targets := []string{svc.Name + ":9090"}
+
+	probe := framework.MakeBasicStaticProbe(group, proberUrl, targets)
+	if _, err := framework.MonClientV1.Probes(ns).Create(context.TODO(), probe, metav1.CreateOptions{}); err != nil {
+		t.Fatal("Creating Probe failed: ", err)
+	}
+
+	p := framework.MakeBasicPrometheus(ns, prometheusName, group, 1)
+	p.Spec.ProbeSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"group": group,
+		},
+	}
+	if _, err := framework.CreatePrometheusAndWaitUntilReady(ns, p); err != nil {
+		t.Fatal(err)
+	}
+
+	if finalizerFn, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, svc); err != nil {
+		t.Fatal(errors.Wrap(err, "creating prometheus service failed"))
+	} else {
+		ctx.AddFinalizerFn(finalizerFn)
+	}
+
+	// Wait for Probe target
+	if err := framework.WaitForTargets(ns, svc.Name, 1); err != nil {
```

Comment:
I think for this test it would be useful to inspect this target a bit further and see that the configuration of the target is as expected.

Reply:
Yeah, you are right. Thanks!